frame/contracts/src/wasm/env_def/macros.rs

```diff
@@ -15,81 +15,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Definition of macros that hides boilerplate of defining external environment
-//! for a wasm module.
-//!
-//! Most likely you should use `define_env` macro.
```

Comment:
Let's delete this module. No macros left.

Reply:
Left it only because of `constrain_closure<R, F>(f: F) -> F` is [still used](https://github.com/paritytech/substrate/blob/ag-macro-rewrite/frame/contracts/proc-macro/src/lib.rs#L390) inside the code expanded by the macro. Haven't found the proper other place to move it to.

Reply:
I think we could get rid of this function if we force the user to write down the return type of the function:
```rust
Result<impl ConvertibleToWasm, TrapReason>
```